modules/top-level/plugins/combine-plugins.nix

```diff
@@ -0,0 +1,102 @@
+{ lib, pkgs, ... }:
```

Comment:
Again, if this isn't a module we should avoid `...`.

We could turn this into a module by having it define an internal option, instead of returning a function. I don't have a strong opinion in this case.